### DISCUSSION TITLE
Client bounding rect for Canvas elements.

### DISCUSSION THREAD

**Discussion by JasonGH17** [leptos-rs/leptos#1695] [Q&A] [answered]
I'm trying to create a simple canvas paint program.
I've tried using the example code provided in the wasm-bindgen guide (1.22) but the line which is being drawn on the canvas doesn't match the cursor's position. (It seems to get farther from the cursor the more the cursor gets farther from canvas coordinates 0,0)
I have a working example using javascript but it requires having the canvas client bounding rect. I can't seem to find a function for getting it in both leptos and wasm-bindgen though...

**Comment by gbj** [comment, marked as answer]
The function you're looking for is probably this one in web-sys: https://docs.rs/web-sys/latest/web_sys/struct.Element.html#method.get_bounding_client_rect

Note that you need to have a web-sys dependency with the DomRect feature enabled; web-sys feature gates most APIs for compile time benefits.

**Reply by JasonGH17** [reply]
Yeah that works! Thank you
